File: eMovieFinder/eMovieFinder.Model/Utilities/TimeStampObject.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace eMovieFinder.Model.Utilities
{
    public class TimeStampObject
    {
        public DateTime? CreationDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
        [ForeignKey("CreatedBy")]
        public int? CreatedById { get; set; }
        public virtual IdentityUser<int> CreatedBy { get; set; }
        [ForeignKey("ModifiedBy")]
        public int? ModifiedById { get; set; }
        public virtual IdentityUser<int> ModifiedBy { get; set; }
    }
    public class TimeStampObjectViewModel
    {
        public DateTime? CreationDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public int? CreatedById { get; set; }
        public int? ModifiedById { get; set; }
        public virtual IdentityUser<int> CreatedBy { get; set; }
        public virtual IdentityUser<int> ModifiedBy { get; set; }
        public string? FormattedCreationDate => CreationDate?.ToString("MMM dd, yyyy");
        public string? FormattedModifiedDate => ModifiedDate?.ToString("MMM dd, yyyy");
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/utilities/time_stamp_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'time_stamp_object.g.dart';

@JsonSerializable()
class TimeStampObject{
  final DateTime? creationDate;
  final DateTime? modifiedDate;
  final int? createdById;
  final int? modifiedById;
  final String? formattedCreationDate;
  final String? formattedModifiedDate;

  TimeStampObject({
    this.creationDate,
    this.modifiedDate,
    this.createdById,
    this.modifiedById,
    this.formattedCreationDate,
    this.formattedModifiedDate
  });

  factory TimeStampObject.fromJson(Map<String, dynamic> json) => _$TimeStampObjectFromJson(json);
  Map<String, dynamic> toJson() => _$TimeStampObjectToJson(this);
}